core/indexer/component.go

```diff
@@ -173,14 +180,30 @@ func checkIndexerState(ctx context.Context) (uint32, error) {
 		return 0, fmt.Errorf("error loading pruning index: %s", err)
 	}
 
-	if !needsToFillIndexer && nodeStatus.GetLedgerPruningIndex() > ledgerIndex {
+	if !needsToFillIndexer && nodeStatus.LedgerIndex < ledgerIndex {
+		CoreComponent.LogInfo("> Network has been reset: indexer index > ledger index")
+		CoreComponent.LogInfo("Re-import initial ledger...")
+
+		if err := deps.Indexer.Clear(); err != nil {
+			return 0, fmt.Errorf("clearing Indexer failed! Error: %w", err)
+		}
+		needsToFillIndexer = true
+		needsCreateProtocolTable = true
+	} else if !needsToFillIndexer && nodeStatus.GetLedgerPruningIndex() > ledgerIndex {
 		CoreComponent.LogInfo("> Node has an newer pruning index than our current ledgerIndex")
 		CoreComponent.LogInfo("Re-import initial ledger...")
 
 		if err := deps.Indexer.Clear(); err != nil {
 			return 0, fmt.Errorf("clearing Indexer failed! Error: %w", err)
 		}
 		needsToFillIndexer = true
+		needsCreateProtocolTable = true
+	}
+
+	if needsCreateProtocolTable {
```

Comment:
Instead of creating the protocol table, we can add that to the state table.

Reply:
alright, I can add it to the state table since only the network id is needed.